### ISSUE TITLE
Helm deployed my chart resources, but fails to record the release so it cannot be managed

### ISSUE THREAD

**Issue by AndrewMorsillo** [helm/helm#4574]
When I helm install my chart all of the resources are created successfully but helm spits out
 
```
NAME:   myapp-prod
Error: getting deployed release "myapp-prod": release: "myapp-prod" not found
```

All of the resources are created correctly and the app is running just fine. After this I cannot manage the release with helm anymore as helm just says "release my app-prod not found".

I believe this started after I upgraded to helm 2.10.0

GKE version

``` 
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.7", GitCommit:"dd5e1a2978fd0b97d9b78e1564398aeea7e7fe92", GitTreeState:"clean", BuildDate:"2018-04-19T00:05:56Z", GoVersion:"go1.9.3", Compiler:"gc", Platfo
rm:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.6-gke.2", GitCommit:"384b4eaa132ca9a295fcb3e5dfc74062b257e7df", GitTreeState:"clean", BuildDate:"2018-08-15T00:10:14Z", GoVersion:"go1.9.3b4", Compiler:"
gc", Platform:"linux/amd64"} 
```

Helm version
``` 
Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
```

This makes it so the only way I can upgrade my release is to manually delete all the resources then run the broken helm install again. Helm upgrade --install has the same behavior.

**Comment by bacongobbler** [comment]
> This makes it so the only way I can upgrade my release is to manually delete all the resources then run the broken helm install again.

Is this intermittent (meaning this happens seemingly random) or consistent (happens every deploy)? Can you check tiller's logs and see what's the output there?

**Comment by AndrewMorsillo** [comment]
Happens every time. Seems that my configmap is too long?

```create: failed to create: ConfigMap "taskhero-prod.v1" is invalid: []: Too long: must have at most 1048576 characters```

That doesn't make sense because I haven't changed my chart since upgrading helm. This worked fine in whatever version I was using previously. Maybe a 1 or 2 line change, but thats it.

Tiller logs
```
[main] 2018/09/01 17:25:42 Starting Tiller v2.10.0 (tls=false)
[main] 2018/09/01 17:25:42 GRPC listening on :44134
[main] 2018/09/01 17:25:42 Probes listening on :44135
[main] 2018/09/01 17:25:42 Storage driver is ConfigMap
[main] 2018/09/01 17:25:42 Max history per release is 0
[tiller] 2018/09/01 17:26:52 preparing install for taskhero-prod
[storage] 2018/09/01 17:26:52 getting release history for "taskhero-prod"
[tiller] 2018/09/01 17:26:52 rendering taskhero chart using values
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/controller-hpa.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/udp-configmap.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/controller-stats-service.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/redis/templates/secrets.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/controller-daemonset.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/redis/templates/pvc.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/tcp-configmap.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/headers-configmap.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/nginx-ingress/templates/controller-metrics-service.yaml" is empty. Skipping.
2018/09/01 17:26:52 info: manifest "taskhero/charts/redis/templates/networkpolicy.yaml" is empty. Skipping.
[tiller] 2018/09/01 17:26:52 performing install for taskhero-prod
[tiller] 2018/09/01 17:26:52 executing 0 crd-install hooks for taskhero-prod
[tiller] 2018/09/01 17:26:52 hooks complete for crd-install taskhero-prod
[tiller] 2018/09/01 17:26:52 executing 0 pre-install hooks for taskhero-prod
[tiller] 2018/09/01 17:26:52 hooks complete for pre-install taskhero-prod
[storage] 2018/09/01 17:26:52 getting release history for "taskhero-prod"
[storage] 2018/09/01 17:26:52 creating release "taskhero-prod.v1"
[storage/driver] 2018/09/01 17:26:56 create: failed to create: ConfigMap "taskhero-prod.v1" is invalid: []: Too long: must have at most 1048576 characters
[tiller] 2018/09/01 17:26:56 warning: Failed to record release taskhero-prod: ConfigMap "taskhero-prod.v1" is invalid: []: Too long: must have at most 1048576 characters
[kube] 2018/09/01 17:26:56 building resources from manifest
[kube] 2018/09/01 17:26:56 creating 19 resource(s)
[tiller] 2018/09/01 17:26:56 executing 0 post-install hooks for taskhero-prod
[tiller] 2018/09/01 17:26:56 hooks complete for post-install taskhero-prod
[storage] 2018/09/01 17:26:56 updating release "taskhero-prod.v1"
[storage/driver] 2018/09/01 17:27:00 update: failed to update: configmaps "taskhero-prod.v1" not found
[tiller] 2018/09/01 17:27:00 warning: Failed to update release taskhero-prod: configmaps "taskhero-prod.v1" not found
[storage] 2018/09/01 17:27:01 getting last revision of "taskhero-prod"
[storage] 2018/09/01 17:27:01 getting release history for "taskhero-prod"
```

**Comment by AndrewMorsillo** [comment]
That is also a pretty cryptic error message. Why wouldn't helm fail and do nothing rather than actually creating my resources and then leaving the release in a broken state?

**Comment by bacongobbler** [comment]
> Too long: must have at most 1048576 characters

That error means that the configmap release data went over the 1MB limit. Usually that means the chart packaged some extra files that doesn't need to be packaged up. Try adding a .helmignore file that removes any files that should not be part of the chart (you can inspect via `helm package`, then inspecting the tarball for what was included in the package). Let me know if that works for you.

I agree that the error's fairly cryptic. Feel like taking a crack at bubbling that error up to the client? :)

**Comment by AndrewMorsillo** [comment]
Looks related to #1413 and #1996

My experience level with helm and Kubernetes is "newbie" so I doubt I would be able to fix this issue properly. From what I understand from the other issues the 1mb limit is an upstream Kubernetes thing. Is that restriction going to be lifted in a future Kubernetes release? It seems a little arbitrary and restrictive

**Comment by AndrewMorsillo** [comment]
I got rid of some stuff I didn't need, brining the chart down to 87kb unzipped and the issue still persists.

**Comment by AndrewMorsillo** [comment]
I ran my install command with --dry-run --debug > debug.txt

The txt file is 2kb. Not sure what is going wrong here.

**Comment by AndrewMorsillo** [comment]
🤦‍♂️ I had accidentally downloaded the fly binary from my concourse installation to my char directly. The fly binary was 15mb. Sorry for the inconvenience! Dumb mistake!

**Comment by texasdave2** [comment]
Wanted to thank Andrew for the clue that tipped me into solving this on my system.

Just like him, everything worked fine until it didn't...   Just wanted to add to this and explain what solved the problem for me.  I was making a bunch of timestamped backups of a values.yaml file each time I did a helm upgrade to the component I was modifying.  After awhile I had dozens of these timestamped backup yaml files that were about 80k each.  They added up to push over the 1 Meg limit, so I no longer keep my backups in that release folder and I deleted a bunch of them - so definitely check the total size of your component chart release folder and see if it is over the limit mentioned in this thread.
